### PR TITLE
[Breaking][Requires Quarkus 3.12+] Use new generated static resources

### PR DESCRIPTION
**Pull request by ia3andy**
With this change Quinoa is finally able to delegate file server to Quarkus Vert.x http using the new GeneratedStaticResourceBuildItem.

- Requires Quarkus 3.12+
- [Breaking] `quarkus.quinoa.index-page` has been removed. Use `quarkus.http.static-resources.index-page` instead.

### CODE REVIEW

**Comment by ia3andy** [comment]
cc @mcruzdev your new stuff again :)

**Comment by ia3andy** [comment]
@melloware since this is for 3.12+ maybe we should start a new branching for backports?

**Comment by melloware** [comment]
I will cut the 2.3.8 release and make a 2.3.X branch so we can start this new version as 2.4.0

**Review by melloware — APPROVED** [review]
LGTM. I just rebased this branch and I made a new 2.3.X branch for 3.8 LTS and I will make main now 2.4.X on for Quarkus 3.12 and higher

File: deployment/src/main/java/io/quarkiverse/quinoa/deployment/QuinoaProcessor.java [hunk 1]
```diff
@@ -1,11 +1,7 @@
 package io.quarkiverse.quinoa.deployment;
 
-import static io.quarkiverse.quinoa.QuinoaRecorder.META_INF_WEB_UI;
-import static io.quarkiverse.quinoa.QuinoaRecorder.QUINOA_ROUTE_ORDER;
 import static io.quarkiverse.quinoa.QuinoaRecorder.QUINOA_SPA_ROUTE_ORDER;
-import static io.quarkiverse.quinoa.deployment.config.QuinoaConfig.isDevServerMode;
-import static io.quarkiverse.quinoa.deployment.config.QuinoaConfig.isEnabled;
-import static io.quarkiverse.quinoa.deployment.config.QuinoaConfig.toHandlerConfig;
+import static io.quarkiverse.quinoa.deployment.config.QuinoaConfig.*;
```
Comment (mcruzdev) [inline deployment/src/main/java/io/quarkiverse/quinoa/deployment/QuinoaProcessor.java:4]: Miss .* import

Comment (melloware) [reply deployment/src/main/java/io/quarkiverse/quinoa/deployment/QuinoaProcessor.java:4]: i am fixing in another PR

**Pull request merged**